CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -361,12 +361,15 @@ public static function validateFinancialAccount($financialAccountId, $financialA
    * with Financial Account
    *
    * @param array $params
+   *   Holds submitted formvalues and params from api for updating/adding contribution
    *
    * @param int $contributionID
+   *   Contribution ID
    *
    * @param obj $form
+   *   Holds form object of CRM_Contribute_Form_Contribution
    *
-   * @return string
+   * @return bool
    *
    */
   public static function checkFinancialTypeHasDeferred($params, $contributionID = NULL, $form = NULL) {
```

Comment:
Okay, I see it now at L393. @colemanw could you comment on what to do here - is this acceptable practice, or should the relevant work to deal with Priceset be done in the form and then passed in to this function?